src/eleveldb_ts.erl

```diff
@@ -18,13 +18,11 @@ encode_key(Elements) when is_list(Elements) ->
 %% binary list stuff is shocking here too
 encode_k2([],                    Bin) -> Bin;
 encode_k2([{timestamp, Ts} | T], Bin) -> encode_k2(T, append(<<Ts:64>>, Bin));
-encode_k2([{float, F}      | T], Bin) -> encode_k2(T, append(<<F:64/float>>,  Bin));
-encode_k2([{integer, I}    | T], Bin) -> encode_k2(T, append(<<I:64/integer>>, Bin));
-encode_k2([{binary, B}     | T], Bin) when is_binary(B) -> 
-    encode_k2(T, append(B, Bin));
-encode_k2([{binary, L}     | T], Bin) when is_list(L) ->
-    B = list_to_binary(L),
-    encode_k2(T, append(B, Bin)).
+encode_k2([{double, F}     | T], Bin) -> encode_k2(T, append(<<F:64/float>>,  Bin));
+encode_k2([{sint64, I}     | T], Bin) -> encode_k2(T, append(<<I:64/integer>>, Bin));
+encode_k2([{varchar, B}    | T], Bin) when is_binary(B) -> encode_k2(T, append(B, Bin));
+encode_k2([{binary, L}     | T], Bin) when is_list(L)   -> B = list_to_binary(L),
```

Comment:
Looking at the tests in riak_ql, a binary type is always a binary and not a list.

https://github.com/basho/riak_ql/blob/develop/src/riak_ql.yrl.tests#L257